reflection.md

```diff
@@ -17,3 +21,9 @@ Write a few brief paragraphs that discuss:
   implementing it? (You don't have to write code for this part. You could say
   "I would use extension to..." or "Scala's rules for valid
   identifiers prevent...")
+
+This design works really well for writing math grammars because the addition operators we added are wrappers around existing functionality to correspond to more math-like formulas. This makes it easy to create representation of things that contains lots of patterns. I can imagine this DSL being used for representing in knitting or embroidered patterns. Stitches can be represented with Characters and the repetition operator. The other operators can be used to show nesting or crossing of stitches. 
+
+Implicit conversion between custom types are significantly more difficult than from a built-in type. Concatenation can also become confusing when many operators are used at once. The telephone example in Program is relatively easy to read because of the simple 3-3-4 structure of US numbers, but more complicated systems like library classifications may require even more syntatic sugar for ease of use.
```

Comment:
I think that implicit conversion get complicated when you have to add extensions to the conversion